internal/devbox/providers/nixcache/setup.go

```diff
@@ -0,0 +1,220 @@
+package nixcache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"time"
+
+	"go.jetpack.io/devbox/internal/envir"
+	"go.jetpack.io/devbox/internal/nix"
+	"go.jetpack.io/devbox/internal/redact"
+	"go.jetpack.io/devbox/internal/setup"
+	"go.jetpack.io/devbox/internal/xdg"
+)
+
+// nixSetupTask adds the user to Nix's trusted-users list so that they can use
+// their private Devbox cache with the Nix daemon.
+type nixSetupTask struct {
+	// username is the OS username to trust.
+	username string
+}
+
+func (n *nixSetupTask) NeedsRun(ctx context.Context, lastRun setup.RunInfo) bool {
+	cfg, err := nix.CurrentConfig(ctx)
+	if err != nil {
+		return true
+	}
+	trusted, _ := cfg.IsUserTrusted(ctx, n.username)
+	if trusted {
+		return false
+	}
+
+	if _, err := nix.DaemonVersion(ctx); err != nil {
+		// This looks like a single-user install, so no need to
+		// configure the daemon.
+		return false
+	}
+	return true
+}
+
+func (n *nixSetupTask) Run(ctx context.Context) error {
+	if os.Getuid() != 0 {
+		return sudo(ctx, n.username)
+	}
+	err := nix.IncludeDevboxConfig(ctx, n.username)
+	if err != nil {
+		return redact.Errorf("modify nix config: %v", err)
+	}
+	return nil
+}
+
+// awsSetupTask configures the OS's root account to authenticate with AWS by
+// obtaining a token from `devbox cache credentials`.
+type awsSetupTask struct {
+	// username is the OS username that the Nix daemon should sudo as when
+	// running `devbox cache credentials`.
+	username string
+}
+
+func (a *awsSetupTask) NeedsRun(ctx context.Context, lastRun setup.RunInfo) bool {
+	return lastRun.Time.IsZero()
+}
```

Comment:
Is aws setup needed for non-daemon? We can just pass env variables into build no? What if user doesn't even have sudo access?

Reply:
Added a check for the Nix daemon here.